### ISSUE TITLE
Long term planning

### ISSUE THREAD

**Issue by nrc** [nrc/dev-tools-team#11]
We've been playing catch-up with IDE support - it is currently one of the most desired features for Rust, but the core community was pretty late in prioritising development. I want to get ahead of the curve and be working on the next important tool before it becomes necessary and overdue. That means getting a better understanding of what is wanted or missing in the current world.

**Comment by Manishearth** [comment]
I think rustfix is the kind of tool that's currently pretty rare in IDEs and has the potential to be amazing.

You can even teach it to fix compile errors, not just lints.

**Comment by killercup** [comment]
Indeed, rustfix has immense potential, especially for people who are learning Rust and can get quick feedback on best practices.

When I say 'rustfix', though, I don't mean just the CLI tool I wrote. The mechanism of how the compiler/clippy supplies suggestions (spans with replacements) can and should be consumed by RLS, so that an editor plugin can apply the fixes directly.
